utils/osm_incremental_cagg_refresh/01-producer.sql

```diff
@@ -0,0 +1,183 @@
+$$This column will be populated with the pid that is currently running this task.
+This allows us to keep track of things, as well as allow us to reschedule an item if
+a worker_pid is no longer active (for whatever reason)$$;
+
+COMMENT ON COLUMN _timescaledb_additional.incremental_continuous_aggregate_refreshes.scheduled IS
+$$To ensure we do actually get to do all the work, the workers will always pick up the
+task that has the lowest priority, and then which one was scheduled first.
+In that way, we have a bit of a priority queue.$$;
+
+-- We want to avoid scheduling the same thing twice, for those tasks that have not yet been
+-- picked up by any worker.
+CREATE UNIQUE INDEX IF NOT EXISTS incr_cagg_refreshes_distinct_tasks_unq ON _timescaledb_additional.incremental_continuous_aggregate_refreshes(
+    continuous_aggregate,
+    window_start,
+    window_end
+) WHERE worker_pid IS NULL AND finished IS NULL;
+
+CREATE INDEX IF NOT EXISTS incr_cagg_refreshes_find_first_work_item_idx ON _timescaledb_additional.incremental_continuous_aggregate_refreshes(
+    priority,
+    scheduled
+) WHERE worker_pid IS NULL;
+
+CREATE INDEX IF NOT EXISTS incr_cagg_refreshes_active_workers_idx ON _timescaledb_additional.incremental_continuous_aggregate_refreshes(
+    worker_pid
+) WHERE worker_pid IS NOT NULL;
+
+-- Calculate the time bucket using the continuous aggregate bucket function configuration
+CREATE OR REPLACE FUNCTION _timescaledb_additional.cagg_time_bucket(INTEGER, TIMESTAMPTZ)
+RETURNS TIMESTAMPTZ AS
+$$
+DECLARE
+    params TEXT[];
+    stmt TEXT;
+    r RECORD;
+    result TIMESTAMPTZ;
+BEGIN
+    SELECT
+        *
+    INTO
+        r
+    FROM
+        _timescaledb_catalog.continuous_aggs_bucket_function
+    WHERE
+        mat_hypertable_id = $1;
+
+    IF NOT FOUND THEN
+        RAISE EXCEPTION 'Continuous Aggregate % not found', $1;
+    END IF;
+
+    params := array_append(params, format('%I => %L::timestamptz', 'ts', $2));
+
+    IF r.bucket_width IS NOT NULL THEN
+        params := array_append(params, format('%I => %L::interval', 'bucket_width', r.bucket_width));
+    END IF;
+
+    IF r.bucket_origin IS NOT NULL THEN
+        params := array_append(params, format('%I => %L::timestamptz', 'origin', r.bucket_origin));
+    END IF;
+
+    IF r.bucket_offset IS NOT NULL THEN
+        params := array_append(params, format('%I => %L::interval', 'offset', r.bucket_offset));
+    END IF;
+
+    IF r.bucket_timezone IS NOT NULL THEN
+        params := array_append(params, format('%I => %L::text', 'timezone', r.bucket_timezone));
+    END IF;
+
+    stmt := format('SELECT time_bucket(%s)', array_to_string(params, ', '));
+    RAISE DEBUG '%', stmt;
+
+    EXECUTE stmt
+    INTO result;
+
+    RETURN result;
+END;
+$$
+LANGUAGE plpgsql STABLE;
+
+CREATE OR REPLACE PROCEDURE _timescaledb_additional.schedule_osm_cagg_refresh(
+    name_mask TEXT DEFAULT '%',
```

Comment:
The procedure should take a schema_name and view_name.
You can make the view_name match a name_mask.


```suggestion
   schema_name TEXT,
    view_name_mask TEXT DEFAULT '%',
```

Reply:
I don't think we need to complicate things even more now... let's get the feedback on Tuesday and if it is really necessary we improve this API.